### ISSUE TITLE
should math include the math delimiters?

### ISSUE THREAD

**Issue by maelle** [ropensci/tinkr#113]
New lines are a constant challenge in babeldown, so I noticed that I was losing the ones after and before the "$$".

I wonder whether the math should include the borders?

``` r
expected <- "example\n\n$$\na_{ij}\n$$\n\n"

temp_file <- withr::local_tempfile()
brio::write_lines(expected, temp_file)
wool <- tinkr::yarn$new(temp_file)
wool$protect_math()
wool$get_protected()
#> {xml_nodeset (3)}
#> [1] <softbreak math="true"/>
#> [2] <text xml:space="preserve" asis="true" math="true">a_{ij}</text>
#> [3] <softbreak math="true"/>
```

<sup>Created on 2024-05-17 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

**Comment by maelle** [comment]
what I mean by a constant challenge: https://github.com/ropensci-review-tools/babeldown/blob/a60a1772fb021f6650e6b64d447d04e86179784d/R/translate.R#L435

**Comment by maelle** [comment]
why does the softbreak have the math attribute but not the asis one?

**Comment by maelle** [comment]
how it looks like with the current tinkr version on main:

``` r
expected <- "example\n\n$$\na_{ij}\n$$\n\n"

temp_file <- withr::local_tempfile()
brio::write_lines(expected, temp_file)
wool <- tinkr::yarn$new(temp_file)
wool$body <- tinkr::protect_math(wool$body)
wool$body |> as.character() |> cat()
#> <?xml version="1.0" encoding="UTF-8"?>
#> <!DOCTYPE document SYSTEM "CommonMark.dtd">
#> <document xmlns="http://commonmark.org/xml/1.0">
#>   <paragraph>
#>     <text xml:space="preserve">example</text>
#>   </paragraph>
#>   <paragraph>
#>     <text xml:space="preserve">$$</text>
#>     <softbreak/>
#>     <text xml:space="preserve" asis="true">a_{ij}</text>
#>     <softbreak/>
#>     <text xml:space="preserve">$$</text>
#>   </paragraph>
#> </document>
```

<sup>Created on 2024-05-17 with [reprex v2.1.0](https://reprex.tidyverse.org)</sup>

here the softbreak doesn't have the math attribute.

**Comment by zkamvar** [comment]
> I wonder whether the math should include the borders?

I think it absolutely should include the borders. Luckily, this is a pretty easy switch. 

> why does the softbreak have the math attribute but not the asis one?

They all have the math attribute, but the softbreaks don't have the asis attributes because for some reason I wrote a helper function that avoided tagging the `softbreak` nodes:

https://github.com/ropensci/tinkr/blob/4c0c6bef58a901deff1122f6ba5e1e594217b69d/R/asis-nodes.R#L31-L33

That being said, I'm thinking it might be better to include the softbreaks in the asis party and math party since it will help filter nodes and it doesn't affect the rendering.

**Comment by maelle** [comment]
I still have a new line problem in https://github.com/ropensci-review-tools/babeldown/pull/70 (the tests fail), I need to investigate.

**Comment by maelle** [comment]
All good now! https://github.com/ropensci-review-tools/babeldown/pull/70

**Comment by maelle** [comment]
Thank you for the change!!

**Comment by zkamvar** [comment]
wunderbar! I've also confirmed this works in pegboard so it looks like I can merge this and move on to the next challenge!

https://github.com/carpentries/pegboard/pull/155/commits/1d1be7bcbd5cc850be39f2e902cb65bcf1184dd6

**Comment by zkamvar** [comment]
fixed in #111
